recipes/fmi4cpp/meta.yaml

```diff
@@ -0,0 +1,48 @@
+{% set name = "fmi4cpp" %}
+{% set version = "0.8.0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/NTNU-IHB/FMI4cpp/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 78616e9c86a23137a8d3a113fe6420207c3f9ea46442e1c75a01215eb2693bb7
+
+build:
+  number: 0
+  # See https://github.com/conda-forge/staged-recipes/pull/18411#issuecomment-1073669556
+  skip: true  # [osx] 
+  run_exports:
+    - {{ pin_subpackage(name, max_pin='x.x.x') }}
+
+requirements:
+  build:
+    - cmake
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - ninja
+  host:
+    - libzip
+    - boost-cpp
+  run:
+    - boost-cpp
```

Comment:
Please look at these warnings from the conda-build link inspector: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=491541&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=06d62fd2-ab49-5add-9ee5-83b6882d7d2b&l=1815

Are you using the header only boost libraries or statically linking? Or sometimes boost is only used for tests? Switch from static to dynamic linking or ensure that the run requirements excludes boost.

Reply:
Only the header-only part of boost-cpp is used, but it is used in the public headers of `fmi4cpp` so if we do not include it, for downstream C++ users `find_package(fmi4cpp REQUIRED)` will fail. If you prefer, I can remove it, but as this is a pure C++ library I tought it was desirable to simplify as much as possible the usage of the library from the point of view of C++ users.

Reply:
You should remove `boost-cpp` from the run requirements because it is not a run requirement, it is a build requirement. AFAIK, conda does not have a `build_exports` mechanism.